access_log/access_log_record.go

```diff
@@ -66,9 +68,16 @@ func (r *AccessLogRecord) makeRecord() *bytes.Buffer {
 	if r.RouteEndpoint == nil {
 		fmt.Fprintf(b, "app_id:MissingRouteEndpointApplicationId")
 	} else {
-		fmt.Fprintf(b, `app_id:%s`, r.RouteEndpoint.ApplicationId)
+		fmt.Fprintf(b, `app_id:%s `, r.RouteEndpoint.ApplicationId)
 	}
 
+	extraHeadersToLog := make(map[string]string)
+	for _, header := range config.ExtraHeadersToLog {
+		extraHeadersToLog[header] = r.FormatRequestHeader(header)
+	}
+	marshalledExtraHeadersToLog, _ := json.Marshal(extraHeadersToLog)
```

Comment:
Hi @shinji62!

> what is the cost of this logging

I don't know to be honest. I will find out. What would be a good measurement for you?
Fire off 100 000 requests in quick succession against current gorouter and against this patched version and note down mem, cpu usage and response time? I think if there should be any negative performance implications it would be from json.Marshal.

> What about a test if config.ExtraHeadersToLo is empty there is no need to make the all process like 

Thats a good idea, presumably we should also disable the logging of the headers: part if the array is empty, this could be done in two ways

Either only check for nil in makeRecord and in the template
```
<% if properties.gorouter.extra_headers_to_log and properties.gorouter.extra_headers_to_log != [] %>
extra_headers_to_log: [ <%= p("extra_headers_to_log").join(", ") %> ]
<% end %>
```
or checking the lenght of the slice in the wrapper if.

```
if config.ExtraHeadersToLog != nil && len(config.ExtraHeadersToLog) != 0 {
...
}
```

I prefer the first suggestion as it removes and extra comparison for each event.